src/Mod/Sketcher/Gui/DrawSketchDefaultHandler.h

```diff
@@ -0,0 +1,1120 @@
+                            // Auto constraining an edge so swap parameters
+                            std::swap(geoId1, geoId2);
+                            std::swap(posId1, posId2);
+                        }
+
+                        auto result = std::find_if(AutoConstraints.begin(),
+                                                   AutoConstraints.end(),
+                                                   [&](const auto& ace) {
+                                                       return ace->Type == Sketcher::Tangent
+                                                           && ace->First == geoId1
+                                                           && ace->Second == ac.GeoId;
+                                                   });
+
+                        // if tangency, convert to point-to-edge tangency
+                        if (result != AutoConstraints.end()) {
+                            (*result)->FirstPos = posId1;
+
+                            if ((*result)->First != geoId1) {
+                                std::swap((*result)->Second, (*result)->First);
+                            }
+                        }
+                        else {
+                            auto c = std::make_unique<Sketcher::Constraint>();
+                            c->Type = Sketcher::PointOnObject;
+                            c->First = geoId1;
+                            c->FirstPos = posId1;
+                            c->Second = geoId2;
+                            AutoConstraints.push_back(std::move(c));
+                        }
+                    } break;
+                    // In special case of Horizontal/Vertical constraint, geoId2 is normally unused
+                    // and should be 'Constraint::GeoUndef' However it can be used as a way to
+                    // require the function to apply these constraints on another geometry In this
+                    // case the caller as to set geoId2, then it will be used as target instead of
+                    // geoId2
+                    case Sketcher::Horizontal: {
+                        auto c = std::make_unique<Sketcher::Constraint>();
+                        c->Type = Sketcher::Horizontal;
+                        c->First = (geoId2 != Sketcher::GeoEnum::GeoUndef ? geoId2 : geoId1);
+                        AutoConstraints.push_back(std::move(c));
+                    } break;
+                    case Sketcher::Vertical: {
+                        auto c = std::make_unique<Sketcher::Constraint>();
+                        c->Type = Sketcher::Vertical;
+                        c->First = (geoId2 != Sketcher::GeoEnum::GeoUndef ? geoId2 : geoId1);
+                        AutoConstraints.push_back(std::move(c));
+                    } break;
+                    case Sketcher::Tangent: {
+                        Sketcher::SketchObject* Obj =
+                            static_cast<Sketcher::SketchObject*>(sketchgui->getObject());

```

Comment:
The Lint, I think it is a false positive. Not sure what can be done other than suppressing it.

Insofar as I understand, if the type where unknown (needing a forward declaration), it would be a compiler error. If it were an access to a member for which the object is unknown, it would be a compiler error.

But let's ask a qualified member...
@wwmayer 

What do you think about this one?

Reply:
My wrong. It is not a mere linter. It is a full fledged compiler warning. It is solved now. Sorry for the noise.

Reply:
A diagnostic error isn't something that can be suppressed but must be fixed.

Often such code still compiles fine because the .cpp file includes the missing class declaration beforehand but the clang code model (i.e. clang-tidy) won't be able to see this because it only parses this header file and header file it includes. If for the required type only a forward declaration is visible the clang code model will fail.